engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -59,42 +59,48 @@ object typing {
     override def objType: TypedClass = TypedClass[Any]
   }
 
-  case class Typed(possibleTypes: Set[TypedClass]) extends TypingResult {
+  // TODO: rename to TypedUnion, ensure that always possibleTypes.size > 1, and cleanup places where was used head/headOption
+  case class Typed(possibleTypes: Set[TypingResult]) extends TypingResult {
```

Comment:
I think we should ensure here that possibleTypes have > 1 element, otherwise we'll have to separate in many places case with possibleTypes.size == 1 from TypedClass?

Reply:
done